src/main/java/com/hotels/bdp/circustrain/bigquery/client/BigQueryMetastoreClient.java

```diff
@@ -152,10 +153,16 @@ public Table getTable(final String databaseName, final String tableName) throws
     ExtractionUri extractionUri = new ExtractionUri();
     ExtractionContainer container = new ExtractionContainer(bigQueryTable, extractionUri, PostExtractionAction.RETAIN);
     extractionService.register(container);
-
-    Table hiveTable = tableServiceFactory.newInstance(new BigQueryToHiveTableConverter().withDatabaseName(databaseName)
-        .withTableName(tableName).withSchema(bigQueryTable.getDefinition().getSchema())
-        .withCols(bigQueryTable.getDefinition().getSchema()).withLocation(extractionUri.getTableLocation()).convert())
+    extractionService.extract();
```

Comment:
OK, so up to now this class was mainly a layer on top of the metastore but it didn't actually do any extractions itself. This method is now actually `extractTable` as opposed to `getTable` but we can't rename it since it's implementing `CloseableMetaStoreClient`. Is there any way to have the extraction somewhere else? Perhaps the schema in the table can be updated later? If this isn't going to be straightforward then this is OK for now, as part of the bigger refactoring we should consider whether it's actually worth having this class implement `CloseableMetaStoreClient` or whether we should split the responsibility out differently.